sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-calculation/e-cert-calculation-process.ts

```diff
@@ -152,13 +170,56 @@ export abstract class ECertCalculationProcess {
     }
   }
 
+  /**
+   * Checks whether the disbursement blocked notification should be created or not.
+   * @param disbursementId disbursement id.
+   * @param entityManager entity manager to execute in transaction.
+   * @returns boolean indicating if the disbursement blocked notification should be created or not.
+   */
+  private async shouldCreateDisbursementBlockedNotification(
+    disbursementId: number,
+    entityManager: EntityManager,
+  ): Promise<boolean> {
+    // Create notifications to the student and ministry when the disbursement is blocked.
+    const notificationMessageId = 16;
+    const notification = await entityManager
+      .getRepository(Notification)
+      .createQueryBuilder("notifications")
+      .select("count(*)::int", "notificationsCount")
+      .addSelect("max(notifications.created_at)", "max_created_at")
+      .innerJoin("notifications.notificationMessage", "notificationMessage")
+      .where("notificationMessage.id = :notificationMessageId", {
+        notificationMessageId,
+      })
+      .andWhere("notifications.permanent_failure_error IS NULL")
```

Comment:
Can you please elaborate on why the `permanent_failure_error` was included in the conditions?

Reply:
I added it because if a notification was not sent because of a `permanent_failure_error`, it should not be counted towards the 3 total max. notifications we would send. Otherwise, it might happen that a notification generated `permanent_failure_error` thrice and hence, a notification never got sent out.

Reply:
`permanent_failure_error` means something that cannot be recovered from. If it falls into that category we should not keep generating other messages.

Reply:
Removed.